Hibernate/OneToOneMapping/Instructor.java

```diff
@@ -0,0 +1,92 @@
+package OneToOneMapping;
+
+import javax.persistence.CascadeType;
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.JoinColumn;
+import javax.persistence.OneToOne;
+import javax.persistence.Table;
+
+@Entity
+@Table(name="instructor")
+public class Instructor {
+
+	@Id
+	@GeneratedValue(strategy=GenerationType.IDENTITY)
+	@Column(name="id")
+	private int id;
+	
+	@Column(name="first_name")
+	private String firstName;
+	
+	@Column(name="last_name")
+	private String lastName;
+
+	@Column(name="email")
+	private String email;
+	
+	@OneToOne(cascade=CascadeType.ALL)
```

Comment:
see what happens when you remove cascade type, use a different CascadeType

Reply:
We will get Transient Object Exception by removing cascade type because we are saving the referenced entities before saving the entity that references.We have other cascade types like merge,refresh,delete,save_update...etc